packages/augur-core/source/contracts/reporting/ShareToken.sol

```diff
@@ -238,15 +245,16 @@ contract ShareToken is ITyped, Initializable, ERC1155, IShareToken, ReentrancyGu
      * @return (uint256 _creatorFee, uint256 _reportingFee) The fees taken for the market creator and reporting respectively
      */
     function sellCompleteSetsForTrade(IMarket _market, uint256 _outcome, uint256 _amount, address _shortParticipant, address _longParticipant, address _shortRecipient, address _longRecipient, uint256 _price, address _affiliateAddress) external returns (uint256 _creatorFee, uint256 _reportingFee) {
-        require(isApprovedForAll(_shortParticipant, msg.sender) == true, "ERC1155: need operator approval to burn short account shares");
-        require(isApprovedForAll(_longParticipant, msg.sender) == true, "ERC1155: need operator approval to burn long account shares");
+        // bypassing approvals for predicate flow
```

Comment:
Wouldn't there need to be some concept of what was approved still?

Reply:
So the predicate sandbox is a controlled environment. You claimed your balances from Matic, however to replay my trade I wouldn't want to approve these contracts again on root. In this sandbox predicate will be able to govern actions.